### PR TITLE
Documentation: add learner in runtime-configuration.md

### PR DESCRIPTION
**Pull request by jingyih**
Updating the `runtime-configuration.md` document to support learner concept.

cc @jpbetz @gyuho @WIZARD-CXY

### CODE REVIEW

**Comment by WIZARD-CXY** [comment]
LGTM

File: Documentation/op-guide/runtime-configuration.md [hunk 1]
```diff
@@ -123,6 +123,45 @@ The new member will run as a part of the cluster and immediately begin catching
 
 If adding multiple members the best practice is to configure a single member at a time and verify it starts correctly before adding more new members. If adding a new member to a 1-node cluster, the cluster cannot make progress before the new member starts because it needs two members as majority to agree on the consensus. This behavior only happens between the time `etcdctl member add` informs the cluster about the new member and the new member successfully establishing a connection to the existing one.
 
+#### Add new member as learner
+
+Starting from v3.4, etcd supports adding new member as learner / non-voting member.
```
Comment (jpbetz) [inline Documentation/op-guide/runtime-configuration.md]: minor grammar nit: "adding a new member"

File: Documentation/op-guide/runtime-configuration.md [hunk 2]
```diff
@@ -123,6 +123,45 @@ The new member will run as a part of the cluster and immediately begin catching
...
+The motivation and design can be found in [design doc](https://etcd.readthedocs.io/en/latest/server-learner.html).
+In order to make the process of adding new member safer,
+and to reduce cluster downtime when the new member is added, it is recommended that the new member is added to cluster
+as learner until it catches up. This can be described as a three step process:
```
Comment (jpbetz) [inline Documentation/op-guide/runtime-configuration.md]: minor grammar nit: "as a learner"

File: Documentation/op-guide/runtime-configuration.md [hunk 3]
```diff
@@ -123,6 +123,45 @@ The new member will run as a part of the cluster and immediately begin catching
...
+ * Add the new member as learner via [gRPC members API][member-api-grpc] or the `etcdctl member add --learner` command.
+ Note that v2 [HTTP member API][member-api] does not support this feature. (If user wants to use HTTP,
+ etcd provides a JSON [gRPC gateway][grpc-gateway], which serves a RESTful proxy that translates HTTP/JSON requests into gRPC messages.)
+
+ * Start the new member with the new cluster configuration, including a list of the updated members (existing members + the new member).
+ This step is exactly the same as before.
+
+ * Promote the newly added learner to voting member via [gRPC members API][member-api-grpc] or the `etcdctl member promote` command.
+ etcd server validates promote request to ensure its operational safety.
+ Only after its raft log has caught up to leader’s can learner be promoted to a voting member.
```
Comment (jpbetz) [inline Documentation/op-guide/runtime-configuration.md:143]: Make it a bit more clear here that it is expected that the promote request/command can fail with a "not ready" type error, in which case the correct thing to do is wait a bit and retry?

**Review by jpbetz — APPROVED** [review]
A couple minor suggestions, grammar nits. I'm okay with this being merged once they've been addressed.

LGTM

**Pull request merged**